tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature

```diff
@@ -315,3 +315,138 @@ Feature: Sharing files and folders with internal users
 #    And file "lorem.txt" should be listed in shared-with-others page on the webUI
     And as "user2" file "lorem (2).txt" should not exist
     But as "user3" file "lorem (2).txt" should exist
+
+  Scenario: share a file & folder with another internal user with as custom role and then set all custom permissions
+    Given user "user2" has logged in using the webUI
+    When the user shares folder "simple-folder" with user "User One" as "Custom Role" using the webUI
+    Then user "User One" should be listed as "Viewer" in the collaborators list for folder "simple-folder" on the webUI
+    When the user sets the custom permissions for "User One" for folder "simple-folder" as:
+      | permission |
+      | delete     |
+      | share      |
+      | create     |
+      | change     |
+    Then user "User One" should be listed as "Custom role" in the collaborators list on the webUI
+    And as "user1" these resources should be listed on the webUI
+      | entry_name                  |
+      | simple-folder (2)           |
+    And these resources should be listed in the folder "simple-folder (2)" on the webUI
+      | lorem.txt |
+    When the user renames file "lorem.txt" to "new-lorem.txt" using the webUI
+    Then file "new-lorem.txt" should be listed on the webUI
+    And file "lorem.txt" should not be listed on the webUI
+    When the user creates a folder with the name "some-folder" using the webUI
+    Then folder "some-folder" should be listed on the webUI
+
+  Scenario Outline: share a file & folder with another internal user and set various custom permission
+    Given user "user2" has logged in using the webUI
+    When the user shares folder "simple-folder" with user "User One" as "Custom Role" using the webUI
+    Then user "User One" should be listed as "Viewer" in the collaborators list for folder "simple-folder" on the webUI
+    When the user sets the custom permissions for "User One" for folder "simple-folder" as:
+      | permission           |
+      | <custom-permission>  |
+    Then user "User One" should be listed as "Custom role" in the collaborators list on the webUI
```

Comment:
check that the correct permissions are shown on the UI for the sharer
check using the API if the correct permissions have been set